clients/pkg/promtail/client/batch.go

```diff
@@ -71,20 +73,36 @@ func (b *batch) add(entry api.Entry) error {
 	return nil
 }
 
-func labelsMapToString(ls model.LabelSet, without ...model.LabelName) string {
-	lstrs := make([]string, 0, len(ls))
-Outer:
+func labelsMapToString(ls model.LabelSet, without model.LabelName) string {
+	var b strings.Builder
+	totalSize := 2
+	lstrs := make([]model.LabelName, 0, len(ls))
+
 	for l, v := range ls {
-		for _, w := range without {
-			if l == w {
-				continue Outer
-			}
+		if l == without {
+			continue
 		}
-		lstrs = append(lstrs, fmt.Sprintf("%s=%q", l, v))
+
+		lstrs = append(lstrs, l)
+		totalSize += len(l) + 2 + len(v) + 3
+	}
+
+	b.Grow(totalSize)
+	b.WriteByte('{')
+	slices.Sort(lstrs)
+	for i, l := range lstrs {
+		if i > 0 {
+			b.WriteString(", ")
+		}
+
+		b.WriteString(string(l))
+		b.WriteString(`="`)
+		b.WriteString(strconv.Quote(string(ls[l])))
```

Comment:
strconv.Quote looks like it creates a string with quotes, the quotes added before and after this end up doubling up on the quotes and making an invalid label string.